### PR TITLE
Provide resolvConf to sandbox container's mounts

### PR DESCRIPTION
**Pull request by darfux**
As https://github.com/kata-containers/runtime/issues/1603 discussed, kata relies on such mount spec to setup resolv.conf for pod VM properly.

Signed-off-by: Li Yuxuan <liyuxuan04@baidu.com>

### CODE REVIEW

**Comment by Random-Liu** [comment]
@darfux Is `/etc/hosts` and `/etc/hostname` needed?

**Comment by darfux** [comment]
@Random-Liu  I think only resolv.conf is ok. `hostname` can be retrieved from ocispec [oci/utils.go#L774](https://github.com/kata-containers/runtime/blob/613fd0fb60a4a88da24012b7a422a30a3f30271d/virtcontainers/pkg/oci/utils.go#L774), while `/etc/hosts` is not considered and using the original VM rootfs version right now.

File: pkg/server/sandbox_run_unix.go [hunk 1]
```diff
@@ -108,6 +108,12 @@ func (c *criService) sandboxContainerSpec(id string, config *runtime.PodSandboxC
 			Type:        "bind",
 			Options:     []string{"rbind", "ro"},
 		},
```
Comment (Random-Liu) [inline pkg/server/sandbox_run_unix.go]: Can you add a comment about why is this needed?

I already forgot why we mounted dev shim, so it would be better to put a comment here.

Comment (darfux) [reply pkg/server/sandbox_run_unix.go]: done, PTAL 😁
